src/routes/safe/store/actions/transactions/__tests__/utils/transactionHelpers.test.ts

```diff
@@ -0,0 +1,875 @@
+  })
+})
+
+describe('isMultiSendTransaction', () => {
+  it('It should return true if given a transaction without value, the data has multisend substring', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: '0x8d80ff0a' })
+
+    // when
+    const result = isMultiSendTransaction(transaction)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return false if given a transaction without data', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: null })
+
+    // when
+    const result = isMultiSendTransaction(transaction)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return true if given a transaction without value, the data has not multisend substring', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: 'thisiswrongdata' })
+
+    // when
+    const result = isMultiSendTransaction(transaction)
+
+    // then
+    expect(result).toBe(false)
+  })
+})
+
+describe('isUpgradeTransaction', () => {
+  it('If should return true if the transaction data is empty', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: null })
+
+    // when
+    const result = isUpgradeTransaction(transaction)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return false if the transaction data is multisend transaction but does not have upgradeTx data', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: '0x8d80ff0a' })
+
+    // when
+    const result = isUpgradeTransaction(transaction)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return true if the transaction data is multisend transaction has upgradeTx data', () => {
+    // given
+    const upgradeTxData = `0x8d80ff0a000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000f200dfa693da0d16f5e7e78fdcbede8fc6ebea44f1cf000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000247de7edef000000000000000000000000d5d82b6addc9027b22dca772aa68d5d74cdbdf4400dfa693da0d16f5e7e78fdcbede8fc6ebea44f1cf00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024f08a032300000000000000000000000034cfac646f301356faa8b21e94227e3583fe3f5f0000000000000000000000000000`
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: upgradeTxData })
+
+    // when
+    const result = isUpgradeTransaction(transaction)
+
+    // then
+    expect(result).toBe(true)
+  })
+})
+
+describe('isOutgoingTransaction', () => {
+  it('It should return true if the transaction has not a to address equal to the safe address and has no empty data', () => {
```

Comment:
It should return true if the transaction recipient is not a safe address and data is not empty